### PR TITLE
Update build clang/llvm/lldb 3.9

### PR DESCRIPTION
**Pull request by mikem8361**
Update build scripts, docs and build pipeline docker images to clang/llvm/lldb 3.9

### CODE REVIEW

**Comment by danmosemsft** [comment]
@mikem8361 this would need shiproom approval. wemight want to hold off at least to check master has no issues.

File: Documentation/building/debugging-instructions.md [hunk 1]
```diff
@@ -98,7 +104,8 @@ However the common commands have been aliased so that you don't need the SOS pre
     clrstack        -> sos ClrStack
     clrthreads      -> sos Threads
     clru            -> sos U
-    dso             -> sos DumpStackObjects
+    createdump      -> sos CreateDump
+	dso             -> sos DumpStackObjects
```
Comment (janvorli) [inline Documentation/building/debugging-instructions.md]: A nit - it seems this line is missing indentation.

**Comment by mikem8361** [comment]
@dotnet-bot test Windows_NT arm Cross Release Build
@dotnet-bot test Windows_NT arm Cross Debug Build

**Comment by sdmaclea** [comment]
test Ubuntu arm64 Cross Release Build

**Comment by sdmaclea** [comment]
test Ubuntu arm64 Debug

**Comment by sdmaclea** [comment]
@dotnet-bot test Ubuntu arm64 Cross Debug Build

**Comment by sdmaclea** [comment]
@dotnet/arm64-contrib I can't trigger Ubuntu arm64 runs.  I have locally verified arm64 builds on my CI system, but it would be nice to run this on dotnet CI too.

**Comment by mikem8361** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test

**Comment by mikem8361** [comment]
@dotnet-bot test CentOS7.1 x64 Release Priority 1 Build and Test
@dotnet-bot test Windows_NT x64 Debug Build and Test

**Comment by mikem8361** [comment]
@dotnet-bot test CentOS7.1 x64 Release Priority 1 Build and Test

**Pull request closed without merging**